File: JavaProjects/StateSolucao03/src/domains/state/CanceladoState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package domains.state;

import domains.Pedido;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author jeffe
 */
public class CanceladoState implements State{
    
    private Pedido pedido;

    public CanceladoState(Pedido pedido) {
        this.pedido = pedido;
    }
    
    @Override
    public void sucessoAoPagar() {
        try {
            throw new Exception("Operação não suportada - pedido cancelado");
        } catch (Exception ex) {
            Logger.getLogger(CanceladoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    @Override
    public void cancelarPedido() {
        try {
            throw new Exception("Operação não suportada - pedido cancelado");
        } catch (Exception ex) {
            Logger.getLogger(CanceladoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    @Override
    public void despacharPedido() {
        try {
            throw new Exception("Operação não suportada - pedido cancelado");
        } catch (Exception ex) {
            Logger.getLogger(CanceladoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
}

File: CsharpProjects/StateSolucao02/StateSolucao02/domains/state/CanceladoState.cs
using StateSolucao01.domains;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StateSolucao02.domains.state
{
    public class CanceladoState : State
    {
        private Pedido pedido;

        public CanceladoState(Pedido pedido)
        {
            this.pedido = pedido;
        }

        void State.cancelarPedido()
        {
            throw new Exception("Operacao não suportada, " +
                "pedido foi cancelado");
        }

        void State.despacharPedido()
        {
            throw new Exception("Operacao não suportada, " +
                "pedido foi cancelado");
        }

        void State.sucessoAoPagar()
        {
            throw new Exception("Operacao não suportada, " +
                "pedido foi cancelado");
        }
    }
}
